crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,279 @@
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+            Ok(SshKey {
+                private_key: private_key_openssh.to_string(),
+                public_key: private_key.public_key().to_string(),
+                key_fingerprint: private_key.fingerprint(HashAlg::Sha256).to_string(),
+            })
+        }
+        KeyType::Rsa => {
+            let private_key: rsa::RsaPrivateKey = match password {
+                Some(password) => {
+                    pkcs8::DecodePrivateKey::from_pkcs8_encrypted_pem(&encoded_key, password)
+                        .map_err(|err| match err {
+                            pkcs8::Error::EncryptedPrivateKey(_) => {
+                                SshKeyImportError::WrongPassword
+                            }
+                            _ => SshKeyImportError::ParsingError,
+                        })?
+                }
+                None => pkcs8::DecodePrivateKey::from_pkcs8_pem(&encoded_key)
+                    .map_err(|_| SshKeyImportError::ParsingError)?,
+            };
+
+            let private_key = ssh_key::private::PrivateKey::from(
+                RsaKeypair::try_from(private_key).map_err(|_| SshKeyImportError::ParsingError)?,
+            );
+            let private_key_openssh = private_key
+                .to_openssh(LineEnding::LF)
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+            Ok(SshKey {
+                private_key: private_key_openssh.to_string(),
+                public_key: private_key.public_key().to_string(),
+                key_fingerprint: private_key.fingerprint(HashAlg::Sha256).to_string(),
+            })
+        }
+        _ => Err(SshKeyImportError::UnsupportedKeyType),
+    }
+}
+
+fn import_openssh_key(encoded_key: String, password: String) -> Result<SshKey, SshKeyImportError> {
+    let private_key = match ssh_key::private::PrivateKey::from_openssh(&encoded_key) {
+        Ok(k) => k,
+        Err(err) => match err {
+            ssh_key::Error::AlgorithmUnknown
+            | ssh_key::Error::AlgorithmUnsupported { algorithm: _ } => {
+                return Err(SshKeyImportError::UnsupportedKeyType)
+            }
+            _ => return Err(SshKeyImportError::ParsingError),
+        },
+    };
+
+    if private_key.is_encrypted() && password.is_empty() {
+        return Err(SshKeyImportError::PasswordRequired);
+    }
+    let private_key = if private_key.is_encrypted() {
+        private_key
+            .decrypt(password.as_bytes())
+            .map_err(|_| SshKeyImportError::WrongPassword)?
+    } else {
+        private_key
+    };
+    let private_key_openssh = private_key
+        .to_openssh(LineEnding::LF)
+        .map_err(|_| SshKeyImportError::ParsingError)?;
+
+    Ok(SshKey {
+        private_key: private_key_openssh.to_string(),
+        public_key: private_key.public_key().to_string(),
+        key_fingerprint: private_key.fingerprint(HashAlg::Sha256).to_string(),
+    })
```

Comment:
**issue:** This is reused in `import_pkcs8_key`. We can replace this with `private_key.try_into()`.